src/board.rs

```diff
@@ -199,9 +242,37 @@ impl Chessboard {
         '.'
     }
 
-    /// Forsyth–Edwards Notation Serializer
-    /// * `chessboard` - The chessboard position to be converted to a FEN.
-    /// # Return: FEN string representing the board's position.
+    /// Serializes a chessboard position into Forsyth–Edwards Notation (FEN).
+    ///
+    /// # Arguments
+    ///
+    /// - `chessboard`: The chessboard position to be converted to FEN.
+    ///
+    /// # Returns
+    ///
+    /// A FEN string representing the board's position.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use kn_o3::Chessboard;
+    ///
+    /// let initial_position = Chessboard::new();
+    /// let fen_string = initial_position.to_string();
+    /// println!("FEN: {}", fen_string);
+    /// ```
+    ///
+    /// The resulting FEN string will consist of six space-separated fields:
+    /// 1. Piece placement
+    /// 2. Whose turn it is ('w' for white, 'b' for black)
+    /// 3. Castling rights
+    /// 4. En passant target square (or "-" if none)
+    /// 5. Halfmove clock (always "0" in this implementation)
+    /// 6. Fullmove number (always "1" in this implementation)
+    ///
+    /// Note: The FEN string is used to represent the state of a chess position in a concise
+    /// and human-readable format. It captures information about piece placement, castling rights,
+    /// en passant targets, and other game state details.
```

Comment:
It's not our job to document what a FEN is, I say replace this with a [more info](https://www.chess.com/terms/fen-chess) link and an example of a FEN string.